CAIPs/caip-xxx.md

```diff
@@ -0,0 +1,126 @@
+---
+
+# CAIP-XX: Chain-Agnostic Token Payment URI
+
+**Title**: Chain-Agnostic Token Payment URI
+**Author**: Pedro Gomes
+**Status**: Draft
+**Type**: Standard
+**Created**: 2025-05-20
+
+---
+
+## Abstract
+
+This proposal defines a URI scheme for token-based payments in a chain-agnostic format. The format encodes the target blockchain, recipient account, token asset, and optional amount using URL-safe syntax and Chain Agnostic Improvement Proposals (CAIPs):
+
+* **CAIP-2** for chain identification
+* **CAIP-19** for token identification (excluding chain prefix)
+* **CAIP-10** for account identification (excluding chain prefix)
+
+---
+
+## Motivation
+
+Digital payments in Web3 are fragmented across wallets, chains, and token standards. Current methods lack consistency and interoperability, especially for use in deep links and QR codes within mobile apps, websites, ecommerce checkouts, or social media posts.
+
+A standardized URI string provides a simple, readable, and composable format for triggering token transfers across different chains using a unified scheme. This enables:
+
+* One-click payments from mobile wallets via deep linking
+* QR code generation for point-of-sale payments
+* Standardized checkout links for ecommerce platforms
+* Tipping or payment links embedded in social media profiles or messages
+* Easy parsing and validation by wallet SDKs, apps, and browsers
+
+---
+
+## Specification
+
+### URI Format
+
+```
+caip:<CHAIN>/pay?recipient=<RECIPIENT>&token=<TOKEN>&amount=<AMOUNT>
+```
+
+* `CHAIN` is **required** and follows CAIP-2
+* `recipient` is **required**, follows CAIP-10 but omits chain prefix
+* `token` is **required**, follows CAIP-19 but omits chain prefix
+* `amount` is **optional**, denominated in the smallest unit of the token (e.g., wei, lamports, satoshi)
```

Comment:
as an integer? as a "json number" (i.e. stringtyped)? does this work for ERC-1155s or other special-snowflake tokens?